addons/mail/models/mail_activity.py

```diff
@@ -795,3 +796,130 @@ def activity_unlink(self, act_type_xmlids, user_id=None):
             domain = ['&'] + domain + [('user_id', '=', user_id)]
         self.env['mail.activity'].search(domain).unlink()
         return True
+
+    def _read_progress_bar(self, domain, group_by, progress_bar):
+        if not (progress_bar['field'] == 'activity_state' and self._fields[group_by].store):
+            return super()._read_progress_bar(domain, group_by, progress_bar)
+
+        # use datetime.utcnow() instead of date.today(), because
+        # it's used in _compute_state_from_date and on mocking in tests
+        today_utc = datetime.utcnow()
+        today = date(year=today_utc.year, month=today_utc.month, day=today_utc.day)
+
+        # Code below is a version of _read_group_raw:
+        # * lazy is always False
+        # * extra joins for mail_activity
+        # * extra grouping by mail_activity.res_id
+        # * extra field in select: days_to_deadline
+        # * extra SELECT wrapper to compute and group by activity_state
+        fields = ['activity_state']
+        # we also group by activity_state, but because it's a virtual field, we'll add it later manually
+        groupby = [group_by]
+        offset = 0
+        limit = None
+        orderby = False
+
+        self.check_access_rights('read')
+        query = self._where_calc(domain)
+
+        groupby_list = groupby
+        annotated_groupbys = [self._read_group_process_groupby(gb, query) for gb in groupby_list]
+        groupby_fields = [g['field'] for g in annotated_groupbys]
+        order = orderby or ','.join([g for g in groupby_list])
+        groupby_dict = {gb['groupby']: gb for gb in annotated_groupbys}
+
+        self._apply_ir_rules(query, 'read')
+        aggregated_fields = []
```

Comment:
Constant, could be passed as keyword-argument right in the function calls of `_read_group_prepare` and `_read_group_fill_temporal` if you care about naming that empty list

Reply:
The same: I would keep it as in original read_group_raw